File: 0165-compare-version-numbers/0165-compare-version-numbers.java
class Solution {
  public int compareVersion(String version1, String version2) {
    List<Integer> subs1 = Arrays.asList(version1.split("[.]")).stream()
      .map(Integer::valueOf)
      .collect(Collectors.toList());
    List<Integer> subs2 = Arrays.asList(version2.split("[.]")).stream()
      .map(Integer::valueOf)
      .collect(Collectors.toList());
    int i = 0;
    while (i < subs1.size() && i < subs2.size()) {
      if (subs1.get(i) < subs2.get(i)) return -1;
      if (subs1.get(i) > subs2.get(i)) return 1;
      i++;
    }
    while (i < subs1.size()) {
      if (subs1.get(i) > 0) return 1;
      i++;
    }
    while (i < subs2.size()) {
      if (subs2.get(i) > 0) return -1;
      i++;
    }
    return 0;
  }
}

File: 0165-compare-version-numbers/0165-compare-version-numbers.py
class Solution:
  def compareVersion(self, version1: str, version2: str) -> int:
    subs1 = [int(d) for d in version1.split(".")]
    subs2 = [int(d) for d in version2.split(".")]
    i = 0
    while i < len(subs1) and i < len(subs2):
      if subs1[i] < subs2[i]: return -1
      if subs1[i] > subs2[i]: return 1
      i += 1
    while i < len(subs1):
      if subs1[i] > 0: return 1
      i += 1
    while i < len(subs2):
      if subs2[i] > 0: return -1
      i += 1
    return 0